src/apps/validation.clj

```diff
@@ -184,11 +184,13 @@
 
 (defn verify-tool-name-version
   [tool]
-  (let [existing-tool (first (select tools (where (select-keys tool [:name :version]))))]
+  (let [existing-tool (first (select tools
+                                     (fields :id :name :version)
+                                     (where (select-keys tool [:name :version]))))]
     (when existing-tool
       (throw+ {:type  :clojure-commons.exception/exists
                :error "A Tool with that name and version already exists."
-               :tool  tool}))))
+               :tool  existing-tool}))))
```

Comment:
There actually is a `clojure-commons.exception-util` function for this one. You could do something like this (assuming the namespace alias is `cxu`):

```clojure
(cxu/exists "A Tool with that name and version already exists." :tool existing-tool)
```